consensus/src/liveness/leader_reputation.rs

```diff
@@ -90,40 +90,61 @@ impl AptosDBBackend {
 
     fn get_from_db_result(
         &self,
+        target_epoch: u64,
         target_round: Round,
         events: &Vec<NewBlockEvent>,
         hit_end: bool,
     ) -> Vec<NewBlockEvent> {
+        let has_larger = events.first().map_or(false, |e| {
+            (e.epoch(), e.round()) >= (target_epoch, target_round)
+        });
+        if !has_larger {
+            // error, and not a fatal, in an unlikely scenario that we have many failed consecutive rounds,
```

Comment:
Can you explain why many failed consecutive rounds will cause this? Is this because history might be pruned?

Reply:
imagine the storage latest round is 10, and we timeout all round from 10 to 100. now we try to decide the leader for 101, we ask for target round of 81=101-20 (exclude_round) but we won't be able to find any rounds >= 81

Reply:
Hmm..how would the system recover from this state? Should we consider changing the semantic to get last N rounds present in storage instead?

Reply:
they'll still agree on the leaders as long as they agree on the same state (round 10)

Reply:
It already does that. 

Issue is that we could be here for two reasons:
- no succesful rounds for a large number of rounds
- node committing too slow, or not having the latest ledger info. In this case, it is going to compute leader incorrectly,  and so we will we see logs about that. So adding this log makes us able to distinguish the other log better - if this log is not present,  but we see logs for wrong proposer,  something bad is happening